### PR TITLE
Fixed the ssl-certs in netbox plugin

### LINKED ISSUE #46542: ssl-certs verification
##### SUMMARY
Netbox plugin gives an error SSL certificate verify failed even after providing validate_certs: False in netbox_inventory.yml file.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
netbox.py inventory plugin

##### ANSIBLE VERSION
```
ansible 2.6.5
  config file = /Users/netbx-ansible/ansible.cfg
  configured module search path = [u'/Users/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Sep  5 2018, 14:52:04) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

##### CONFIGURATION

```
[defaults]
host_key_checking = False
inventory_plugins  = /Users/d071242/netbx-ansible

[inventory]
enable_plugins = netbox
```

##### OS / ENVIRONMENT
~/ansible> sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G65

##### STEPS TO REPRODUCE
Used below mentioned details in netbox_inventory.yml file
Command used is "ansible-inventory -v --list -i netbox_inventory.yml"

```
name: netbox
plugin_type: inventory
author:
    - Remy Leone (@sieben)
    - Anthony Ruhier (@Anthony25)
short_description: NetBox inventory source
description:
    - Get inventory hosts from NetBox

plugin: netbox
api_endpoint: https://netbox.xyz
validate_certs: False
token:
    required: False
group_by:
  - device_roles
query_filters:
  - role: network-edge-router
strict_permissions: False
```

##### EXPECTED RESULTS
It should be used as a dynamic inventory to list the hosts from Netbox.

##### ACTUAL RESULTS
```
~/netbx-ansible> ansible-inventory -vvv --list -i netbox_inventory.yml
ansible-inventory 2.6.5
  config file = /Users/netbx-ansible/ansible.cfg
  configured module search path = [u'/Users/d071242/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible-inventory
  python version = 2.7.15 (default, Sep  5 2018, 14:52:04) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
Using /Users/netbx-ansible/ansible.cfg as config file
Fetching: https://netbox.xyz/api/dcim/sites/?limit=0
Fetching: https://netbox.xyz/api/dcim/regions/?limit=0
Fetching: https://netbox.xyz/api/tenancy/tenants/?limit=0
Fetching: https://netbox.xyz/api/dcim/racks/?limit=0
Fetching: https://netbox.xyz/api/dcim/device-roles/?limit=0
Fetching: https://netbox.xyz/api/dcim/device-types/?limit=0
Fetching: https://netbox.xyz/api/dcim/manufacturers/?limit=0
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/netbx-ansible/netbox.py", line 249, in refresh_tenants_lookup
    tenants = self.get_resource_list(api_url=url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
SSLValidationError: Failed to validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible, /usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726).

Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/netbx-ansible/netbox.py", line 254, in refresh_racks_lookup
    racks = self.get_resource_list(api_url=url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
SSLValidationError: Failed to validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible, /usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726).

Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/netbx-ansible/netbox.py", line 239, in refresh_sites_lookup
    sites = self.get_resource_list(api_url=url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
SSLValidationError: Failed to validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible, /usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726).

Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/netbx-ansible/netbox.py", line 244, in refresh_regions_lookup
    regions = self.get_resource_list(api_url=url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
SSLValidationError: Failed to validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible, /usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726).

Exception in thread Thread-5:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/netbx-ansible/netbox.py", line 259, in refresh_device_roles_lookup
    device_roles = self.get_resource_list(api_url=url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
SSLValidationError: Failed to validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible, /usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726).

Exception in thread Thread-6:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/netbx-ansible/netbox.py", line 264, in refresh_device_types_lookup
    device_types = self.get_resource_list(api_url=url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
SSLValidationError: Failed to validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible, /usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726).

Exception in thread Thread-7:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/netbx-ansible/netbox.py", line 269, in refresh_manufacturers_lookup
    manufacturers = self.get_resource_list(api_url=url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
SSLValidationError: Failed to validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible, /usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726).

Fetching: https://netbox.xyz/api/dcim/devices/?limit=0&role=network-edge-router
 [WARNING]:  * Failed to parse /Users/netbx-ansible/netbox_inventory.yml with netbox plugin: Failed to
validate the SSL certificate for netbox.xyz:443. Make sure your managed systems have a valid CA
certificate installed. You can use validate_certs=False if you do not need to confirm the servers identity but this
is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/ansible,
/usr/local/etc/openssl. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed
(_ssl.c:726).

  File "/usr/local/lib/python2.7/site-packages/ansible/inventory/manager.py", line 270, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/Users/netbx-ansible/netbox.py", line 374, in parse
    self.main()
  File "/Users/netbx-ansible/netbox.py", line 349, in main
    hosts_list = self.fetch_hosts()
  File "/Users/netbx-ansible/netbox.py", line 312, in fetch_hosts
    return self.get_resource_list(self.device_url)
  File "/Users/netbx-ansible/netbox.py", line 162, in get_resource_list
    api_output = self._fetch_information(api_url)
  File "/Users/netbx-ansible/netbox.py", line 137, in _fetch_information
    response = open_url(url, headers=self.headers, timeout=self.timeout)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 958, in open_url
    r = urllib_request.urlopen(*urlopen_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 427, in open
    req = meth(req)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 773, in http_request
    build_ssl_validation_error(self.hostname, self.port, paths_checked, e)
  File "/usr/local/lib/python2.7/site-packages/ansible/module_utils/urls.py", line 579, in build_ssl_validation_error
    raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))

 [WARNING]: Unable to parse /Users/netbx-ansible/netbox_inventory.yml as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

{
    "_meta": {
        "hostvars": {}
    },
    "all": {
        "children": [
            "ungrouped"
        ]
    },
    "ungrouped": {}
}
```

### PR DESCRIPTION
**Pull request by nikkytub**
##### SUMMARY
Fixed the ssl-certs verification error in netbox plugin 

Fixed #46542

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
netbox inventory plugin

##### ANSIBLE VERSION
```
ansible 2.6.5
  config file = /Users/netbx-ansible/ansible.cfg
  configured module search path = [u'/Users/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Sep  5 2018, 14:52:04) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

##### ADDITIONAL INFORMATION
```
~/netbx-ansible> ansible-inventory -vvv --list -i netbox_inventory.yml
{
    "_meta": {
        "hostvars": {
            "sw-xyz": {
                "device_roles": [
                    "ACI Leaf"
                ],
                "device_types": [
                    "xyz"
                ],
                "manufacturers": [
                    "xyz"
                ],
                "racks": [
                    "xyz"
                ],
                "sites": [
                    "xyz"
                ],
                "tenants": [
                    "abc"
                ]
            }
        }
    },
    "all": {
        "children": [
            "sites_region-name",
            "ungrouped"
        ]
    },
    "sites_region-name": {
        "hosts": [
            "region-specific"
        ]
    },
    "ungrouped": {}
}
```

### CODE REVIEW

File: lib/ansible/plugins/inventory/netbox.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ class InventoryModule(BaseInventoryPlugin):
     NAME = 'netbox'
 
     def _fetch_information(self, url):
-        response = open_url(url, headers=self.headers, timeout=self.timeout)
+        response = open_url(url, headers=self.headers, timeout=self.timeout, validate_certs=False)
```
Comment (ericwb) [inline lib/ansible/plugins/inventory/netbox.py]: Why is certificate validation being turned off?  Why not use the validate_certs parameter to the module?  I'd expect to see changes to parse() to honor the validate_certs argument and pass to open_url.  The examples should also be updated to include one example with this parameter.

**Review by ericwb — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/inventory/netbox.py (hunk 1 above)
Comment (nikkytub) [reply lib/ansible/plugins/inventory/netbox.py]: Done. Please have a look at the latest commit e993e41a220a21693ab80a4ec9b7c9f637b230d2

File: lib/ansible/plugins/inventory/netbox.py [hunk 2]
```diff
@@ -43,6 +43,7 @@
                 - sites
                 - tenants
                 - racks
+                - tags
```
Comment (ericwb) [inline lib/ansible/plugins/inventory/netbox.py]: This appears to be an unrelated change.  You should probably put in a separate pull request.

File: lib/ansible/plugins/inventory/netbox.py [hunk 3]
```diff
@@ -172,44 +174,45 @@ def group_extractors(self):
...
+            "tags": self.extract_tags,
             "device_roles": self.extract_device_role,
             "device_types": self.extract_device_type,
             "manufacturers": self.extract_manufacturer
         }
 
     def extract_device_type(self, host):
         try:
-            return self.device_types_lookup[host["device_type"]["id"]]
+            return [self.device_types_lookup[host["device_type"]["id"]]]
```
Comment (ericwb) [inline lib/ansible/plugins/inventory/netbox.py]: Why did the returns here and below have to change?

File: lib/ansible/plugins/inventory/netbox.py [hunk 4]
```diff
@@ -362,6 +369,7 @@ def parse(self, inventory, loader, path, cache=True):
         token = self.get_option("token")
         self.api_endpoint = self.get_option("api_endpoint")
         self.timeout = self.get_option("timeout")
+        self.validate_certs = self.get_option("validate_certs")
```
Comment (ericwb) [inline lib/ansible/plugins/inventory/netbox.py:372]: Is get_option automatically defaulting the value to True when unset?

File: lib/ansible/plugins/inventory/netbox.py [hunk 5]
```diff
@@ -371,4 +379,4 @@ def parse(self, inventory, loader, path, cache=True):
         # Filter and group_by options
         self.group_by = self.get_option("group_by")
         self.query_filters = self.get_option("query_filters")
-        self.main()
+        self.main()
```
Comment (ericwb) [inline lib/ansible/plugins/inventory/netbox.py]: Gotta remove that end line character to please the CI

**Review by ericwb — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/inventory/netbox.py (hunk 4 above)
Comment (nikkytub) [reply lib/ansible/plugins/inventory/netbox.py:372]: Yes

File: lib/ansible/plugins/inventory/netbox.py (hunk 3 above)
Comment (nikkytub) [reply lib/ansible/plugins/inventory/netbox.py]: Thats not my change. I copied the latest changes instead of syncing my fork

File: lib/ansible/plugins/inventory/netbox.py (hunk 5 above)
Comment (nikkytub) [reply lib/ansible/plugins/inventory/netbox.py]: ok)

File: lib/ansible/plugins/inventory/netbox.py (hunk 4 above)
Comment (nikkytub) [reply lib/ansible/plugins/inventory/netbox.py:372]: Please review the changes @ericwb

**Review by ericwb — APPROVED** [review]
LGTM

**Comment by nikkytub** [comment]
@sieben @Anthony25 @FragmentedPacket Please review the changes.

**Comment by FragmentedPacket** [comment]
LGTM

**Comment by remyleone** [comment]
LGTM

**Comment by remyleone** [comment]
LGTM

**Comment by remyleone** [comment]
@gundalow Could we merge it?

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.8.

Thanks you for the PR and reviews!

**Comment by gundalow** [comment]
@nikkytub Hi I believe this was your first PR to be merged into Ansible. Congratulations!

I'm randomly selecting some PRs to see how it was for you, with the aim of making it easier for future contributors. As a new contributor I'm particularly interested as you don't have any prior knowledge of the Ansible process.

1. Generally speaking, how was the process for you? 
2. Did it make sense what to do to progress the PR? 
3. Did you read any of the developer (`dev_guide`) or contributor documentation?
4. What in the documentation was useful or confusing, missing or could be improved
5. Did you know you could run `ansible-test` locally?
6. What areas would you be interested in contributing to?


https://github.com/ansible/community/issues/353

**Comment by nikkytub** [comment]
@gundalow Hi, thanks for this welcoming message)
1. The process was easy for me. 
2. Yes I knew what to do to progress it.
3. No I didn't read it. However, I will read it soon.
4. I haven't read it yet so can't say anything about it.
5. Yes
6. I like Ansible and think of it as a powerful tool. I would like to contribute to Netbox and openstack related topics at the moment. In future might consider something else as well.

**Comment by gundalow** [comment]
@nikkytub Thanks for the feedback. I see one other `netbox` PR in flight https://github.com/ansible/ansible/pull/46936
